stable/blackbox-exporter/values.yaml

```diff
@@ -0,0 +1,41 @@
+## number of exporter instances
+##
+apiVersion: apps/v1
+# or apps/v1beta2
+# or apps/v1beta1
+
+restartPolicy: Always
+
+user: prometheus
+# set to a crypted password, such as can be generated like:
+# openssl passwd -crypt <password>
+# For testing, 'FAJzAuOSlGGT6' is 'password'
+password:
+
+image:
+  repository: prom/blackbox-exporter
+  tag: v0.11.0
+  pullPolicy: IfNotPresent
+proxyImage:
```

Comment:
Why do you have this proxy image configured here? Rather than expose trough nginx you should create an ingress object and allow the users to customise it

Reply:
If you want to protect this resource either you use `networkPolicies` or a proxy like oauth configured[ trough annotations](https://github.com/kubernetes/ingress-nginx/tree/master/docs/examples/external-auth)

Reply:
The resource is protected by password inside the cluster.

Reply:
You can also use networkPolicies or oauth proxies with this chart. The use of the fronting nginx is optional. For sites that want to spread around blackbox exporters so they can increase their visibility, all they need is this chart set to net=host and k8s/helm on a single node. No ingress needed. Monitoring should be as simple as possible.

Reply:
I still discourage you to run a sidecar to protect blackbox, again: you can achieve the same results with ingress and networkPolicies. To me it's an anti pattern because k8s already solved this problem for you.

Reply:
I agree with @gianrubio. The proxy should not be part of the chart. I'd rather suggest you make this configurable. Here's an example:

https://github.com/kubernetes/charts/blob/86c217302513fed924cfd57808fbc9b51c031623/stable/nginx-ingress/values.yaml#L187-L219

Reply:
I understand what @kfox1111 is saying here, but I do also think this is a bit of an anti-pattern. Based on your argument about keeping it simple, what's the point of running monitoring systems on Kubernetes at all, when you could just run something directly on the host VMs?

That said, given that this is disabled by default and not a very complex configuration, I think it may be _okay_. I would just add comments above the value here and in the README to explain exactly what and how this proxyImage is used (e.g. only used if you set a password).

I think @unguiculus' approach is a good one too, you could always include a separate values.yaml file that includes the nginx basic auth example.